cmd/checkmetrics/csv.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/csv"
+	"errors"
+	"math"
+	"os"
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/montanaflynn/stats"
+)
+
+// csvRecord holds the data for a single 'test'. Some of the data is imported
+// from the CSV file, and other fields are then calculated from that data
+type csvRecord struct {
+	Records [][]string // The record slices imported from the file
+
+	// All items below here are generated or calculated from the Records
+	ResultStrings []string  // Hold the array of all the Results as strings
+	Results       []float64 // Result array converted to floats
+	Iterations    int       // How many results did we gather
+	Mean          float64   // The 'average'
+	MinVal        float64   // Smallest value we saw
+	MaxVal        float64   // Largest value we saw
+	SD            float64   // Standard Deviation
+	CoV           float64   // Co-efficient of Variation
+}
+
+// load reads in a CSV 'Metrics' results file from the file path given
```

Comment:
I think it would be useful to see an example of a valid, minimal CSV results file somewhere (either here, or in the `README`).